src/cast_core/include/cast_core/http.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_7A9C2F9E_ECCA_4AC9_AF39_8EE877217977_INCLUDED
+#define HEADER_7A9C2F9E_ECCA_4AC9_AF39_8EE877217977_INCLUDED
+
+#include <string>
+
+#include <boost/log/trivial.hpp>
+#include <boost/throw_exception.hpp>
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/ssl.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl/error.hpp>
+#include <boost/asio/ssl/stream.hpp>
+
+namespace genny::actor::http {
+
+boost::beast::ssl_stream<boost::beast::tcp_stream> make_tls_stream(std::string host, std::string port) {
+    namespace beast = boost::beast; // from <boost/beast.hpp>
+    namespace http = beast::http;   // from <boost/beast/http.hpp>
+    namespace net = boost::asio;    // from <boost/asio.hpp>
+    namespace ssl = net::ssl;       // from <boost/asio/ssl.hpp>
+    using tcp = net::ip::tcp;       // from <boost/asio/ip/tcp.hpp>
+
+    net::io_context ioc;
+
+    // The SSL context is required, and holds certificates
+    ssl::context ctx(ssl::context::tlsv11);
+
+#if defined(WIN32) || defined(_WIN32) || defined(__WIN32) && !defined(__CYGWIN__)
+    // This does not work on Windows. If we ever care,
+    // see https://stackoverflow.com/a/40710806
```

Comment:
It seems like the SO question has the solution though. Did that not work?
It seems reasonable to leave this as-is since we don't run CI on Windows or readily have Windows dev boxes at our disposal. But if include additional context about why we didn't pursue this it will help future maintainers come to the same "yup not worth it" conclusion that we did here.